src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/CodeActions/RazorFullyQualifiedCodeActionTranslator.ts

```diff
@@ -13,13 +13,44 @@ export class RazorFullyQualifiedCodeActionTranslator implements IRazorCodeAction
     public applyEdit(
         uri: vscode.Uri,
         edit: vscode.TextEdit): [vscode.Uri | undefined, vscode.TextEdit | undefined] {
-        // The edit for this should just translate without additional help.
-        throw new Error('Method not implemented.');
+        // This is kind of wrong. We're manually trimming whitespace and adjusting
+        // for multi-line edits that're provided by O#. Right now, we do not support multi-line edits
+        // (ex. refactoring code actions) however there are certain supported edits which O# is automatically
+        // formatting for us (ex. FullyQualifiedNamespace) into multiple lines, when it should span a single line.
+        // This is due to how we render our virtual cs files, with fewer levels of indentation to facilitate
+        // appropriate error reporting (if we had additional tabs, then the error squigly would appear offset).
+        //
+        // The ideal solution for this would do something like:
+        // https://github.com/dotnet/aspnetcore-tooling/blob/4c8dbd0beb073e6dcee33d96d174453bf44d938a/
+        // src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/DefaultRazorFormattingService.cs#L264
+        // however we're going to hold off on that for now as it isn't immediately necessary and we don't
+        // (currently) support any other kind of multi-line edits.
```

Comment:
@ajaybhargavb with your latest formatting changes, would this now be obsolete?

Reply:
No. What the comment says still holds true. Only that link might need to get updated.

Reply:
It is now [here](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CodeBlockDirectiveFormattingPass.cs#L143) which might change again soon 😆

Reply:
Thanks for the clarification! 😄